src/include/sof/audio/buffer.h

```diff
@@ -83,12 +85,24 @@ extern struct tr_ctx buffer_tr;
 #define BUFF_PARAMS_RATE	BIT(2)
 #define BUFF_PARAMS_CHANNELS	BIT(3)
 
-/* audio component buffer - connects 2 audio components together in pipeline */
+struct comp_buffer;
+
+/*
+ * audio component buffer - connects 2 audio components together in pipeline.
+ *
+ * The buffer is a hot structure that must be shared on certain cache
+ * incoherent architectures.
+ *
+ * Access flow
+ * 1) Lock acquired from using uncache ptr.
+ * 2) Invalidate after lock acquired.
+ * 3) safe to use cached pointer for access.
+ * 4) Writeback cached data
+ * 5) release lock using uncache pointer.
```

Comment:
@lgirdwood This seems pretty neat. We have quite a few buffer_lock users (50+ in latest main), so I wonder if this should be via a new API. I.e. old users would get uncached buffer objects and existing code would continue to work.

To get more performance,. components can opt-in to use a new API to get cached buffer objects (and basicly using the usage pattern you outline here).

Reply:
Yeah, @ranj063 and I were discussing the naming last night as this will need done for pipeline and comp_dev atm. Im thinking of
```c
struct thing *thing_acquire(struct thing *thing);

struct thing *thing_release(struct thing *thing);
```

Will update for pipeline and compdev.